lib/block-parser.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+if (!class_exists('Gutenberg_Block_Parser_State', false)):
+
+class Gutenberg_Block_Parser_State {
+	public $block_stack;
+}
+
+endif;
+
+if (!class_exists('Gutenberg_Block_Parser', false)):
+
+class Gutenberg_Block_Parser {
+	const BLOCK_COMMENT_OPEN  = '(^<!--)';
+	const BLOCK_COMMENT_CLOSE = '(^/?-->)';
+	const BLOCK_NAME          = '(^[[:alpha:]](?:[[:alnum:]]|/[[:alnum:]])*)i';
```

Comment:
@nb I don't think that we should totally avoid RegExps. In this case I'm only using simple RegExps to tokenize and not to parse. As @nylen pointed out we unfortunately can't limit this to known block types because it has to parse everything.

The grammar doesn't use RegExps and we wouldn't have to here, but I felt like it was going to be faster and clearer than running character-by-character the way the PEG parser does.